src/shared/Microsoft.Git.CredentialManager/StringExtensions.cs

```diff
@@ -25,5 +25,108 @@ public static bool IsTruthy(this string str)
+        public static bool IsFalsey(this string str)
+        {
+            return StringComparer.OrdinalIgnoreCase.Equals(str, bool.FalseString) ||
+                   StringComparer.OrdinalIgnoreCase.Equals(str, "0") ||
+                   StringComparer.OrdinalIgnoreCase.Equals(str, "off") ||
+                   StringComparer.OrdinalIgnoreCase.Equals(str, "no");
+        }
+
+        /// <summary>
+        /// Check if the string is considered to be either 'truthy' or 'falsey' (values considered equivalent to 'true' and 'false').
+        /// </summary>
+        /// <remarks>
+        /// Git considers several different values to be equivalent to 'true' and 'false'; we try to be consistent with this
+        /// behavior.
+        /// <para/>
+        /// See the following Git documentation for a list of values considered to be equivalent to 'true' and 'false':
+        /// https://git-scm.com/docs/git-config#git-config-boolean
+        /// </remarks>
+        /// <param name="str">String value to check.</param>
+        /// <returns>True if the value is 'truthy', 'false' if the value is 'falsey', null otherwise.</returns>
+        public static bool? ToBooleany(this string str)
+        {
+            if (IsTruthy(str))
+            {
+                return true;
+            }
+
+            if (IsFalsey(str))
+            {
+                return false;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Check if the string is considered to be either 'truthy' or 'falsey' (values considered equivalent to 'true' and 'false').
+        /// </summary>
+        /// <remarks>
+        /// Git considers several different values to be equivalent to 'true' and 'false'; we try to be consistent with this
+        /// behavior.
+        /// <para/>
+        /// See the following Git documentation for a list of values considered to be equivalent to 'true' and 'false':
+        /// https://git-scm.com/docs/git-config#git-config-boolean
+        /// </remarks>
+        /// <param name="str">String value to check.</param>
+        /// <param name="defaultValue">Default value to return if the string is neither 'truthy', 'falsey', or has not been set.</param>
+        /// <returns>True if the value is 'truthy', 'false' if the value is 'falsey', <paramref name="defaultValue"/> otherwise.</returns>
+        public static bool ToBooleanyOrDefault(this string str, bool defaultValue)
+        {
+            return ToBooleany(str) ?? defaultValue;
+        }
+
+        /// <summary>
+        /// Trim all characters at the end of the string from the last index of the given character.
+        /// </summary>
+        /// <param name="str">String to trim.</param>
+        /// <param name="c">Character to locate the index of.</param>
+        /// <returns>Trimmed string.</returns>
+        public static string TrimEndIndexOf(this string str, char c)
```

Comment:
This and the subsequent `TrimStartIndexOf` don't make it explicit exactly what they do. Maybe `TruncateFromLastIndexOf` and `TruncateUntilFirstIndexOf` or something similar?

Reply:
Also add to the docstring whether the character is included or excluded from the result?

Reply:
:+1: